### PR TITLE
feat: add contribution RFC

### PR DESCRIPTION
**Pull request by 0xPoe**
## Summary

A contribution plugin will be developed for tichi, which is mainly responsible for labeling external members' PRs with `contribution` or `first-time-contributor` labels.

## Links

- [Full Rendered Proposal](https://github.com/hi-rustin/rfcs/blob/rustin-patch-contribution/active-rfcs/0001-contribution.md)

### CODE REVIEW

File: active-rfcs/0001-contribution.md
Comment (andylokandy) [inline active-rfcs/0001-contribution.md:33]: No need to distinguish between external and internal author in this case?

File: active-rfcs/0001-contribution.md
Comment (andylokandy) [inline active-rfcs/0001-contribution.md:24]: Is `org member` equivalent to committer?

File: active-rfcs/0001-contribution.md
Comment (0xPoe) [reply active-rfcs/0001-contribution.md:33]: I think it may may not be necessary, because the contribution will also be added according to whether it is members, we only need to distinguish whether he is the first contribution.

File: active-rfcs/0001-contribution.md
Comment (0xPoe) [reply active-rfcs/0001-contribution.md:24]: Yes.

Comment (andylokandy) [reply active-rfcs/0001-contribution.md:24]: Ok

**Review by andylokandy — APPROVED** [review]

**Comment by 0xPoe** [comment]
/assign @Mini256 

PTAL~ Thanks!

File: active-rfcs/0001-contribution.md
Comment (0xPoe) [reply active-rfcs/0001-contribution.md:24]: Sorry, it's reviewers.

File: active-rfcs/0001-contribution.md
Comment (Mini256) [inline active-rfcs/0001-contribution.md:25]: If I remember correctly, this API must only judge public members.

Comment (0xPoe) [reply active-rfcs/0001-contribution.md:25]: I don't think so.

Please refer to :
- https://docs.github.com/en/rest/reference/orgs#check-organization-membership-for-a-user
- https://docs.github.com/en/rest/reference/orgs#check-public-organization-membership-for-a-user

Comment (0xPoe) [reply active-rfcs/0001-contribution.md:25]: We use the first one.

Comment (Mini256) [reply active-rfcs/0001-contribution.md:25]: > We use the first one.

Maybe it depends on the way of authorization. If we visit the URL directly, it will redirect to the second API.

Comment (0xPoe) [reply active-rfcs/0001-contribution.md:25]: Sorry, I don't understand you .

Comment (Mini256) [reply active-rfcs/0001-contribution.md:25]: Without permission to access private members, access `https://api.github.com/orgs/tikv/members/Mini256`,
will be redirected to `https://api.github.com/organizations/41004122/public_members/Mini256`

Comment (0xPoe) [reply active-rfcs/0001-contribution.md:25]: What is the result of bringing authorization?

Comment (0xPoe) [reply active-rfcs/0001-contribution.md:25]: I think the reason for this redirect is because only internal members can see private members.

**Comment by 0xPoe** [comment]
/hold

Need more discussion

File: active-rfcs/0001-contribution.md
Comment (0xPoe) [reply active-rfcs/0001-contribution.md:25]: I have tested it on postman using token. It works fine.

image

Comment (Mini256) [reply active-rfcs/0001-contribution.md:25]: Got it!

**Review by Mini256 — APPROVED** [review]

File: active-rfcs/0001-contribution.md
Comment (andylokandy) [reply active-rfcs/0001-contribution.md:24]: why not to use the sig json?

Comment (0xPoe) [reply active-rfcs/0001-contribution.md:24]: Because I think even active-contributors need to get some attention.

Comment (0xPoe) [reply active-rfcs/0001-contribution.md:24]: I remember you used to think so too. 🤣

Comment (andylokandy) [reply active-rfcs/0001-contribution.md:24]: I mean, use the committer in the sig json as the filter condition.

Comment (0xPoe) [reply active-rfcs/0001-contribution.md:24]: I don't think the reviewer's pr should be labeled, because reviewers are generally familiar with the collaborative process and can find others to help with reviews as quickly as possible.

Also by definition, it is more obvious and better to understand by whether it is an org member to classify whether it is a contribution or not.

**Pull request merged**
